### PR TITLE
feat((pin 262) crr dd): [sc-24352] Improve Human Readable ID

### PR DESCRIPTION
**Pull request by reecebrend**
Story details: https://app.shortcut.com/wmfs/story/24352

### CODE REVIEW

File: lib/components/state-resources/reset-sequence/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const _ = require('lodash')
+
+class ResetSequence {
+  init (resourceConfig, env) {
+    this.id = `${_.snakeCase(resourceConfig.namespace)}.${_.snakeCase(resourceConfig.id)}`
+    this.client = env.bootedServices.storage.client
+  }
+
+  async run (event, context) {
+    await this.client.query(`ALTER SEQUENCE ${this.id} RESTART WITH 1;`)
```
Comment (meenahoda) [inline lib/components/state-resources/reset-sequence/index.js:10]: Is it worth adding some checks to make sure it's an actual sequence from a blueprint component?

Comment (reecebrend) [reply lib/components/state-resources/reset-sequence/index.js]: Added in most recent commit

File: lib/components/state-resources/reset-sequence/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+class ResetSequence {
+  init (resourceConfig, env) {
+    this.id = `${_.snakeCase(resourceConfig.namespace)}.${_.snakeCase(resourceConfig.id)}`
+    this.client = env.bootedServices.storage.client
+  }
+
+  async run (event, context) {
+    try {
+      await this.client.query(`ALTER SEQUENCE ${this.id} RESTART WITH 1;`)
```
[the remark this answers is not in the document]
Comment (reecebrend) [reply lib/components/state-resources/reset-sequence/index.js]: Addressed in most recent commits

File: lib/components/state-resources/get-current-value-from-sequence/index.js [hunk 3]
```diff
@@ -2,13 +2,23 @@ const _ = require('lodash')
...
     this.prefix = resourceConfig.prefix || ''
     this.client = env.bootedServices.storage.client
+    this.storage = env.bootedServices.storage
   }
 
   async run (event, context) {
-    const value = await this.client.query(`SELECT last_value FROM ${this.id}`)
+    const seq = await this.storage.checkSequenceExists(this.id)
+    if (!seq) {
+      context.sendTaskFailure({
```
Comment (meenahoda) [inline lib/components/state-resources/get-current-value-from-sequence/index.js:15]: You need a `return` here

File: lib/components/state-resources/get-next-value-from-sequence/index.js [hunk 4]
```diff
@@ -2,13 +2,23 @@ const _ = require('lodash')
...
     this.prefix = resourceConfig.prefix || ''
     this.client = env.bootedServices.storage.client
+    this.storage = env.bootedServices.storage
   }
 
   async run (event, context) {
-    const value = await this.client.query(`select nextval('${this.id}')`)
+    const seq = await this.storage.checkSequenceExists(this.id)
+    if (!seq) {
+      context.sendTaskFailure({
```
Comment (meenahoda) [inline lib/components/state-resources/get-next-value-from-sequence/index.js:15]: You need a `return` here

File: lib/components/state-resources/reset-sequence/index.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    this.sqlId = `${_.snakeCase(resourceConfig.namespace)}.${_.snakeCase(resourceConfig.id)}`
+    this.id = `${resourceConfig.namespace}_${resourceConfig.id}`
+    this.client = env.bootedServices.storage.client
+    this.storage = env.bootedServices.storage
+  }
+
+  async run (event, context) {
+    const seq = await this.storage.checkSequenceExists(this.id)
+    if (!seq) {
+      context.sendTaskFailure({
```
Comment (meenahoda) [inline lib/components/state-resources/reset-sequence/index.js:14]: You need a `return` here

**Pull request merged**